salt/vault/roles.sls

```diff
@@ -0,0 +1,18 @@
+{% set roles = salt.pillar.get('vault:roles') %}
+{% for role_id, role in roles.items() %}
+{% for environment in role.environments %}
+{% set environment = salt.pillar.get('environments:{}'.format(environment), {}) %}
+  {% for purpose in env_info.purposes %}
```

Comment:
That's coming from the different environment blocks in the `environment_settings.sls` pillar file. For example:
```yaml
environments:
  mitx-qa:
    purposes:
      current-residential-live: mitx/dogwood.3-2rc2
      current-residential-draft: mitx/dogwood.3-2rc2
...
```